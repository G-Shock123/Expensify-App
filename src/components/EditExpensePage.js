import React from 'react'
import {connect} from 'react-redux'
import ExpenseFrom from './ExpenseForm'
import {startEditExpense,startRemoveExpense } from '../actions/expenses'
import RemoveModal from './RemoveModal'


export class EditExpensePage extends React.Component{
    state={
        selectedOption:undefined
    }

    handleNoChange = () =>{
        this.setState(() => ({ selectedOption: false }))
        console.log(this.state.selectedOption);
        
    }

    handleConfirmRemove = () =>{
        this.setState(() => ({ selectedOption: false }))
        this.onRemove()
        
    }
    onStartRemove = () =>{

        this.setState(() => ({ selectedOption: true }))
        console.log(this.state.selectedOption);
        
    }
    onRemove = () =>{
       
        
        this.props.startRemoveExpense({id:this.props.expense.id})
        this.props.history.push('/')

    }
    onSubmit = (expense)=>{
        console.log(expense);
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/')
    }
    render(){
        return(
            <div>
             <div className="page-header">
                <div className ="content-container">
                 <h1 className="page-header__title">Edit Expense</h1>
                </div>
             </div>
             <div className ="content-container">
             
            
                <ExpenseFrom
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}

                />
                <button className="button button--secondary "
                onClick={this.onStartRemove}


                > Remove Expense </button>
                </div>

                <RemoveModal 
                    selectedOption={this.state.selectedOption}
                    handleNoChange={this.handleNoChange}
                    handleConfirmRemove={this.handleConfirmRemove}
                
                />
            </div>

        )
    }


}

const mapDispatchToProps = (dispatch,props)=>{
    return{
        startEditExpense: (id, expense) => dispatch(startEditExpense(id,expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
    }
}
    
const mapStateToProps = (state, props)=>{
    return{
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage)