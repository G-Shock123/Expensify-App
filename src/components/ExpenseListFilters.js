import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter} from '../actions/filters'
import{sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

// you confuse calender with calendar occasionally

export class ExpenseListFilters extends React.Component{
    state= {
        calenderFocused: null
    }
    onDatesChange = ({startDate, endDate}) =>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)

    }
    onFocusChange = (calenderFocused) =>{
        this.setState(()=>({calenderFocused}))
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    

    }

    onSelectChange = (e) => {
        if (e.target.value === 'amount'){
            this.props.sortByAmount()
            console.log(e.target.value);
            

        }else if(e.target.value ==='date'){
            this.props.sortByDate()
            console.log(e.target.value);
        }
        

    }
    render(){
        return(
            <div className="content-container">
            <div className="input-group"> 
                <div className="input-group__item">
                        <input
                            className="text-input"
                            placeholder="Search Expenses"
                            type="text"
                            value={this.props.filters.text} onChange={this.onTextChange}
                        />
                </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            onChange={this.onSelectChange}
                            value={this.props.filters.sortBy}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calenderFocused}
                            onFocusChange={this.onFocusChange}
                            isOutsideRange={() => { false }}
                            numberOfMonths={1}
                            showClearDates={true}
                        />
                </div>
               
            
            </div>
             

                
               

            </div>

        )
    }
}


const mapStateToProps = (state) =>{
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch)=> ({
    setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    setEndDate:(endDate) => dispatch(setEndDate(endDate)),
    setTextFilter :(value)=>dispatch(setTextFilter(value)),
    sortByAmount:()=> dispatch(sortByAmount()),
    sortByDate:()=>dispatch(sortByDate())



})
export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters) 