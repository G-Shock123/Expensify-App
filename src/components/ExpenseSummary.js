import React from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total' 
import numeral from 'numeral'

export class ExpenseSummary extends React.Component{
    
    
    render(){
        const word = this.props.expenses.length===1?'thing':'things'
        const hiddenAmount = this.props.totalExpenses.length - this.props.expenses.length
        const hiddenText = hiddenAmount=== 1? ` hidden expense not shown as a result of filters.`:`hidden expenses not shown as a result of filters.`
        
        return(
            
            <div>
               <div className="page-header">
                <div className="content-container">
                        <h1 className="page-header__title"> You are looking at <span>{this.props.expenses.length}</span> {word}. The total is <span>{numeral(getExpensesTotal(this.props.expenses) / 100).format('$0,0.00')}</span></h1> 
                        <p className="page-header__paragraph">  <span>{hiddenAmount}</span> {hiddenText} </p>
                        <div className="page-header__actions">
                            <Link className="button" to ="/create">Add Expense</Link>
                        
                        </div>
                    </div>
                </div>
          </div>
        )
    }
}


const mapStateToProps = (state)=>{
    return{
        expenses: selectExpenses(state.expenses, state.filters),
        totalExpenses:state.expenses
    }
}

export default connect(mapStateToProps)(ExpenseSummary)