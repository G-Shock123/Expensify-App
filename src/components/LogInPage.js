import React from 'react'
import {connect} from 'react-redux'
import {startGoogleLogIn, startFacebookLogIn} from '../actions/auth'


export const LogInPage = ({startGoogleLogIn, startFacebookLogIn}) => (
    <div className ="box-layout"> 

      <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>Get expenses under control</p>
            <button className= "button" onClick={startGoogleLogIn}>Login with Google</button>
            <div className="box-layout__spacing"></div>
            <button className= "button" onClick={startFacebookLogIn}>Login with Facebook</button>
      </div>
        
    
    </div>
)

const mapDispatchToProps = (dispatch) =>({
    startGoogleLogIn: ()=>dispatch(startGoogleLogIn()),
    startFacebookLogIn:()=>dispatch(startFacebookLogIn())
})

export default connect(undefined,mapDispatchToProps)(LogInPage)