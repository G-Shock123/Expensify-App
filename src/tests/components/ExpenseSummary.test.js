import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import {ExpenseSummary} from '../../components/ExpenseSummary'
import altFilters from '../fixtures/filters'



test('Making sure rendered properly when everthing is there ',()=>{
    const wrapper = shallow(<ExpenseSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()

})

test('Making sure rendered properly when everthing is there ', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} filters={altFilters}/>)
    expect(wrapper).toMatchSnapshot()

})