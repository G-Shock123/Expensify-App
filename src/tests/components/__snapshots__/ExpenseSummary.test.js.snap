// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Making sure rendered properly when everthing is there  1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
         You are looking at 
        <span>
          3
        </span>
         
        things
        . The total is 
        <span>
          $451.95
        </span>
      </h1>
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`Making sure rendered properly when everthing is there  2`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
         You are looking at 
        <span>
          3
        </span>
         
        things
        . The total is 
        <span>
          $451.95
        </span>
      </h1>
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;
